craft_application/services/project.py

```diff
@@ -0,0 +1,353 @@
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""A service for handling access to the project."""
+
+from __future__ import annotations
+
+import copy
+import os
+import pathlib
+from typing import TYPE_CHECKING, Any, cast, final
+
+import craft_parts
+import craft_platforms
+from craft_cli import emit
+
+from craft_application import errors, grammar, util
+from craft_application.models.grammar import GrammarAwareProject
+
+from . import base
+
+if TYPE_CHECKING:
+    from craft_application import models
+    from craft_application.application import AppMetadata
+
+    from .service_factory import ServiceFactory
+
+
+class ProjectService(base.AppService):
+    """A service for handling access to the project."""
+
+    __platforms: dict[str, craft_platforms.PlatformDict] | None
+    __project_file_path: pathlib.Path | None
+    __raw_project: dict[str, Any] | None
+    _project_dir: pathlib.Path
+    _project_model: models.Project | None
+
+    def __init__(
+        self, app: AppMetadata, services: ServiceFactory, *, project_dir: pathlib.Path
+    ) -> None:
+        super().__init__(app, services)
+        self.__platforms = None
+        self.__project_file_path = None
+        self.__raw_project: dict[str, Any] | None = None
+        self._project_dir = project_dir
+        self._project_model = None
+
+    def resolve_project_file_path(self) -> pathlib.Path:
```

Comment:
Please review these docstrings both as-is and in the context of the [autodocs](https://canonical-craft-application--646.com.readthedocs.build/en/646/reference/services/project.html#api-documentation).